app/Decsys/Repositories/Contracts/IComponentRepository.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using Decsys.Data.Entities;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Decsys.Repositories.Contracts
+{
+    interface IComponentRepository
+    { 
+        Component Create(int id, Guid pageId, string type);
```

Comment:
these interfaces need to return DTO's (Models) not Entities, because the Entities are tied to the data provider (e.g. LiteDb) but the interfaces need to be provider agnostic so they can be used by the service layer.

Hopefully there are models for most cases already in `Models/` but you might need to create some.